### PR TITLE
added continent support (#3)

### PR DESCRIPTION
**Pull request by salvogs**

### CODE REVIEW

File: sample_config.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
 	"ignored_ports": [ 123 ]
     },
     "countries": {
-       "whitelist": ["IT", "DE", "CH", "NL"],
+       "whitelist": ["IT", "DE", "CH", "NL",
+                     "AF", "AS", "EU", "NA", "OC", "SA", "AN"],
```
Comment (lucaderi) [inline sample_config.json:11]: How do you plan to specify the continent in the config file? I don't think it's wise to mix countries with continents, what do you think?

File: Configuration.cpp [hunk 2]
```diff
@@ -136,14 +136,23 @@ bool Configuration::readConfigFile(char *path) {
...
   u_int16_t id = country2u16(country);
   std::unordered_map<u_int16_t, Marker>::iterator it = countries.find(id);
 
-  if(it == countries.end())
-    return(default_marker); /* Not found */
-  else
+  if(it != countries.end())
     return(it->second);
+
+  /* if country not found check continent */
```
Comment (lucaderi) [inline Configuration.cpp:146]: Good idea

File: Configuration.h [hunk 3]
```diff
@@ -46,7 +46,7 @@ class Configuration {
   inline void setQueueId(int nfq_id)                        { nfq_queue_id = nfq_id;  }
   inline void setCountryMarker(u_int16_t country, Marker m) { countries[country] = m; }
   inline Marker getDefaultMarker()                          { return(default_marker); }
-  Marker getCountryMarker(char *country);
+  Marker getCountryMarker(char *country, char *continent);
```
Comment (lucaderi) [inline Configuration.h:49]: Good but in this case the method name is misleading, please rename it.

**Review by lucaderi — CHANGES_REQUESTED** [review]

File: sample_config.json (hunk 1 above)
Comment (salvogs) [reply sample_config.json:11]: Yes, it's true. I should specify continents in a separate json object "continents"

**Comment by lucaderi** [comment]
Closing PR as of https://github.com/ntop/ipt_geofence/pull/11

**Pull request closed without merging**